agot-bg-game-server/src/common/ingame-game-state/game-data-structure/westeros-card/WinterIsComingWesterosCardType.ts

```diff
@@ -12,16 +12,17 @@ export default class WinterIsComingWesterosCardType extends WesterosCardType {
             westerosDeckI: currentDeckI
         });
 
-        const deck = westerosGameState.game.westerosDecks[currentDeckI];
+        let deck = westerosGameState.game.westerosDecks[currentDeckI];
+        const dontShuffle = Math.max(0,westerosGameState.game.revealedWesterosCards - 1);
 
         // Reset discarded state
         deck.forEach(card => { card.discarded = false; });
```

Comment:
That is problematic for round 9. In case of WiC Next Westeros Cards will again show 3 cards so I guess we not only have to filter for discarded cards we also have to calculate `remainingWesterosCardCount = maxRounds - actualRound`.